lib/audio_pool.dart

```diff
@@ -47,18 +47,21 @@ class AudioPool {
       await player.setVolume(volume);
       await player.resume();
 
-      StreamSubscription<void> subscription;
+      late StreamSubscription<void> subscription;
 
       final Stoppable stop = () {
         _lock.synchronized(() async {
-          final AudioPlayer p = currentPlayers.remove(player.playerId);
-          subscription?.cancel();
-          await p.stop();
-          if (availablePlayers.length >= maxPlayers) {
-            await p.release();
-          } else {
-            availablePlayers.add(p);
+          if (currentPlayers.containsKey(player.playerId)) {
+            final AudioPlayer p = currentPlayers.remove(player.playerId)!;
```

Comment:
to avoid the `!` you can do:
```
  localVar = currentPlayers.remove(player.playerId)
  if (localVar != null) {
    ..
  }
```
contains doesn't play as nice with NS

Reply:
Not sure I prefer that, if I do that, we will have one addition variable, while here we just have an `!` at the end of the remove.

Reply:
Or use `dartlin` :-) .

```dart
currentPlayers.remove(player.playerId).let((it) {
    ...
});
```

Reply:
it's less characters but it's a potential flaw

Reply:
also you check for the presence twice, once in contains and then again at remove.
variables are free!